controllers/external-accounts.js

```diff
@@ -69,6 +71,14 @@ const linkExternalAccount = async (req, res) => {
       userId
     );
 
+    try {
+      const unverifiedRoleId = config.get("discordUnverifiedRoleId");
+      await discordServices.removeRoleFromUser(unverifiedRoleId, attributes.discordId, req.userData);
+    } catch (error) {
+      logger.error(`Error getting external account data: ${error}`);
+      return res.status(500).json({ message: `Role Deletion failed. Please contact admin.` });
+    }
+
```

Comment:
So since there a 500 respone code, below. Why do we need to create it at multiple places?
What handling is done here or how is it different from the below serviceUnavailable one?

Reply:
Okay cool. 
But please use res.boom instead

Reply:
I've made the changes. Can you please check it?